app.py

```diff
@@ -169,7 +169,25 @@ def dashboard():
         return render_template('dashboard.html', Equipments = users)
     flash("No Equipments available.", "danger")
     return render_template('home.html')
-    
+
+#flask route to view user profile with user history
+@app.route('/profile/<string:username>')
+@is_logged_in
+def profile(username):
+    cur = mysql.connection.cursor()
+    result = cur.execute("SELECT UserId, username, name, accountType FROM users WHERE username = %s", [username])
+    cur_user = cur.fetchone()
+    UserId = cur_user['UserId']
+    result = cur.execute(
+        "SELECT e.Name as equipmentName,\
+            b.fromDateTime, b.toDateTime, b.RequestStatus,\
+            b.SName, b.SEmail from Bookings b\
+            INNER JOIN Equipments e\
+            ON e.id = b.EquipID WHERE b.UserId = %s", [UserId])
+    history = cur.fetchall()
+    cur.close()
+    return render_template('profile.html', history = history, user=cur_user)
```

Comment:
How is this different from /user/<userID>?

Reply:
This is exactly the same. Why redundant code?

Reply:
I think the right way to address #11 is to use `/user/<userId` route, but change the `@is_admin` check to a new decorator `@is_same_user`. Right @Manvi07 ?

Reply:
There needs to be a different page, as the admin will have the authority to update the request status and not the user.

Reply:
So that part can be included using `if else` in the `.html` file?

Reply:
Why do we need to create a different page and a different route?

Reply:
In the user function, only the admin can access the data but in this function current session user can access their own data. I tried using userId but I was having issues passing the variable into the navbar file because I cannot directly use userId as a variable.

Reply:
I understand what you are saying. But, in this case, we can just work with the route `/profile` instead of `/profile/username` right? Can you change the code to make it like this?

Reply:
I have changed the url as /profile

Reply:
When?

Reply:
What you are currently using is the URL `/profile/<username>` right? 

What I am suggesting is to just have `/profile` and get the username from the `session['username']`.

Reply:
I have committed those changes do I have to create a ner pull request?

Reply:
No, you don't need to create a new Pull Request. The changes look good. I'll test them tomorrow and then merge. Good work @yashikabaan 👍